pkg/internal/kube/store.go

```diff
@@ -36,10 +36,27 @@ func qName(om *informer.ObjectMeta) qualifiedName {
 	return qualifiedName{name: om.Name, namespace: om.Namespace, kind: om.Kind}
 }
 
-// MetaSourceLabels allow overriding some metadata from kubernetes labels
-type MetaSourceLabels struct {
-	ServiceName      string `yaml:"service_name" env:"BEYLA_KUBE_META_SOURCE_LABEL_SERVICE_NAME"`
-	ServiceNamespace string `yaml:"service_namespace" env:"BEYLA_KUBE_META_SOURCE_LABEL_SERVICE_NAMESPACE"`
+// MetadataSources allow overriding some metadata from kubernetes labels and annotations
+type MetadataSources struct {
+	ServiceNameAnnotations      []string `yaml:"service_name_annotations" env:"BEYLA_KUBE_ANNOTATIONS_SERVICE_NAME" envSeparator:","`
+	ServiceNamespaceAnnotations []string `yaml:"service_namespace_annotations" env:"BEYLA_KUBE_ANNOTATIONS_SERVICE_NAMESPACE" envSeparator:","`
+	ServiceNameLabels           []string `yaml:"service_name_labels" env:"BEYLA_KUBE_LABELS_SERVICE_NAME" envSeparator:","`
+	ServiceNamespaceLabels      []string `yaml:"service_namespace_labels" env:"BEYLA_KUBE_LABELS_SERVICE_NAMESPACE" envSeparator:","`
+}
+
+var DefaultMetadataSources = MetadataSources{
+	ServiceNameAnnotations: []string{
+		"resource.opentelemetry.io/service.name",
```

Comment:
all annotations starting with "resource.opentelemetry.io/" should be supported 

if you want a flag (not sure that it's needed) then just "useAnnotationsForResourceAttributes"

Reply:
Yeah, this PR currently changes what we already support: service name, namespace and instance ID. Service version and deployment environment will go into another PR, as they are attributes that aren't currently provided by Beyla.

I'd prefer not using the `useAnnotationsForResourceAttributes` flag, as it seems a very OTEL-opinionated option and I'd prefer something more flexible to use in other environments and use case.

Reply:
OK, then maybe make it more generic like https://github.com/grafana/beyla/pull/1415#discussion_r1861807372